### ISSUE TITLE
Missing Ticket Content from Mail

### ISSUE THREAD

**Issue by brucegibbins** [django-helpdesk/django-helpdesk#1063]
When a ticket is raised via email that has multiple parts with the body being html the content is not used for ticket Description or Comment. 

I suspect that this may be somehow related to the email being an O365 mailbox.

I have troubleshot the code and can see where the multipart sections are parsed in the _extract_part_data()_ function. In my case the content type is 'text/html'.

This portion of the code populates the _part_full_body_ variable after parsing with beautifulsoup but it does not populate the _part_body_ variable as it does if the message was pure text.

Because of this the calling function _object_from_message()_ does not receive any value for the _body_ variable. The end result is that the following conditions looking for _body_ content all fail.

I am not sure if this is an RFC formatting issue. But I am working on trying to resolve unless someone has any ideas

**Comment by uhurusurfa** [comment]
The code will need to be upodated to handle the described situation.
If you are able to fix yourself and raise a PR please feel free otherise it will gt fixed in due cxourse as develoipers are able to devote the time to it.

**Comment by brucegibbins** [comment]
Thank you for your reply.

I am considering the PR option. However I have made a number of changes
including the ability to use Oauth for mail authenticaton.

Therefore, there a number of "changes' spread across a number of modules.

What would be the best way to manage this in the context of a PR?

Thanks in advance.

**Comment by uhurusurfa** [comment]
Do you have an example email where this is happening?
It sounds like it is an amail that does not have a text/plain part to it and would make it quicker to create a unit test for this if you can procide the sample that is breaking it..

**Comment by brucegibbins** [comment]
Hi. I actually think now that the small change I made here has led to the other issue I posted regarding the extra line feed after the body was extracted. Let me not waste too much of your time and get quicktests working with my OAUTH mod and then i can relook at this original post. Kine Regards

**Comment by brucegibbins** [comment]
I am working on a PR for this. 

If the mail body is html which in our case, it is because most staff use Outlook. 

I have discovered that the after determining that the mail part subtype is 'html' the variable _full_part_body_ is initialised but the variable _part_body_ never carries a value when this condition is true.  Then in the calling function _object_from_message()_ the variable body never carries a value and then this is used to populate the description field downstream. Therefore, the description remains empty. If there are multiple mail parts and one of these happens to be text, then the _part_body_ variable does get a value, but it may not be the one expected.

Something needs to happen near line 778 in in email.py unless I am missing something.

**Comment by brucegibbins** [comment]
Yep. I put some changes in on ourside and was getting better results. But
tests on main side failed.

For me as I need a solution in relatively quick time but would like to see
contribution back may I suggest the following as a small step forward that
can be ignored by others who may get source from repo or pypi.

* introduce new queue email method type. Lets call it for what it is O365

* it uses generic oauth2 library but need to store client id and access
token guff so maybe that goes in external file or settings. It could go
into table but then need migrations and security

* then i put new pickup code in a new email module or it can go into
existing email.py. either way it becomes avaialble for cron job or in my
case schedule task

* seperate test written

The thing is that I can get 100 accuracy for me by using an existing o365
module that handles all of the authentication and all of the o365 mail
fluff with attachments etc.

Whilst this may be a pip install dependency so is pinax. So really a
concern in my opinon and maybe if configuration driven can be done in a non
break fashion

I am sure you have a much broader view of none breaking and so i have
probably offended you. But in some cases and in my opinion sometimes better
to break and fix. I think if the project did have reliable o365 integration
there may be more uptake and more contriution.

**Comment by uhurusurfa** [comment]
#1104 should fix this issue - please confirm and close if so.
